src/main/java/picard/sam/SamToFastq.java

```diff
@@ -285,7 +290,7 @@ private File makeReadGroupFile(final SAMReadGroupRecord readGroup, final String
         }
         fileName = IOUtil.makeFileNameSafe(fileName);
         if (preExtSuffix != null) fileName += preExtSuffix;
-        fileName += ".fastq";
+        fileName += COMPRESS_OUTPUTS_PER_RG ? ".fastq.gz" : ".fastq";

```

Comment:
As it is currently, the new option only allows to set the compression for read-group files, as only for these the suffix name cannot be chosen on the command line. By contrast, if the (UNPAIRED|SECOND_END|FIRST_END)_FASTQ is set to a filename with a suffixed `.gz`, the output file will be gzipped automatically. So the current solution is just minimally invasive.

But of course I can add a USE_COMPRESSED_OUTPUT option, but then conflicts between the output file names chosen explicitly with or without `.gz` suffix and the new option need to be resolved somehow.

If the filename is suffixed with `.gz` and USE_COMPRESSED_OUTPUT==false (the default), I would compress, for backward compatibility.

More problematic is the second situation, if the filename is *not* suffixed with `.gz` but USE_COMPRESSED_OUTPUT==true.

We could compress, which would kind of break backward compatibility in the sense that the filename is not decisive, but as the user chooses the new option he/she apparently (?) is aware what he/she is doing and really wants to have compression, no matter what the output file suffix is.

We could also leave the output uncompressed, which -- in total -- would implement a logic of the kind: filename suffix has higher priority than compression toggle.

I leave that choice to you guys.